src/chainstate/burn/db/sortdb.rs

```diff
@@ -619,6 +645,67 @@ impl <'a> SortitionHandleTx <'a> {
         self.set_stacks_block_accepted_at_tip(&chain_tip, consensus_hash, parent_stacks_block_hash, stacks_block_hash, stacks_block_height)
     }
 
+    fn pick_recipients(&self, reward_set_vrf_seed: &SortitionHash, next_pox_info: Option<&RewardCycleInfo>) -> Result<Option<RewardSetInfo>, BurnchainError> {
+        if let Some(next_pox_info) = next_pox_info {
+            if let PoxAnchorBlockStatus::SELECTED_AND_KNOWN(ref anchor_block, ref reward_set) = next_pox_info.anchor_status {
+                if reward_set.len() == 0 {
+                    return Ok(None)
+                }
+
+                let chosen_recipients = reward_set_vrf_seed.choose(OUTPUTS_PER_COMMIT.try_into().unwrap(),
+                                                                   reward_set.len().try_into().unwrap());
+                Ok(Some(RewardSetInfo {
+                    anchor_block: anchor_block.clone(),
+                    recipients: chosen_recipients.into_iter().map(|ix| {
+                        let recipient = reward_set[ix as usize].clone();
+                        (recipient, u16::try_from(ix).unwrap())
+                    }).collect()
+                }))
+            } else {
+                Ok(None)
+            }
+        } else {
+            let last_anchor = SortitionDB::get_last_anchor_block_hash(self, &self.context.chain_tip)?;
+            if let Some(anchor_block) = last_anchor {
+                // known 
+                // get the reward set size
+                let reward_set_size = self.get_reward_set_size()?;
+                if reward_set_size == 0 {
+                    Ok(None)
+                } else {
+                    let chosen_recipients = reward_set_vrf_seed.choose(OUTPUTS_PER_COMMIT.try_into().unwrap(),
+                                                                       reward_set_size as u32);
+                    let mut recipients = vec![];
+                    for ix in chosen_recipients.into_iter() {
+                        let ix = u16::try_from(ix).unwrap();
+                        let recipient = self.get_reward_set_entry(ix)?;
+                        recipients.push((recipient, ix));
+                    }
+                    Ok(Some(RewardSetInfo {
+                        anchor_block,
+                        recipients }))
+                }
+            } else {
+                // no anchor block selected
+                Ok(None)
+            }
+        }
+    }
+
+    fn get_reward_set_entry(&self, entry_ix: u16) -> Result<StacksAddress, db_error> {
```

Comment:
when manipulating reward set entries, shouldn't we be dealing with bitcoin addresses, instead of stack addresses?

Reply:
We already use `StacksAddress` to represent both addresses on the Stacks chain, as well as addresses on the burnchain (for example, see `BurnchainRecipient` -- it uses a `StacksAddress`).

Reply:
Yep -- I used StacksAddress here because it's used in a bunch of places to represent both addresses. Not sure if that's the best thing for us to be doing, but would prefer to deal with that separately from this PR.